CHANGELOG.md

```diff
@@ -13,6 +13,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fixed index bound and extended to the surface layer 
```

Comment:
This comment is insufficient.    Bug fixes should give enough description that someone experiencing the bug might be able to notice the fix in the change log.   (After I review the remaining changes I may come back with a suggestion.)

Reply:
Still not sure I understand what is actually fixed here.   But a better description would hint at something like "Fixed problem in treatment of vertical regridding ..."

Reply:
if it is non-flip, the 3rd dimension of plx may start from 0 too.  The codes after that assume it starts from 1. That is the problem

Reply:
There are 2 bugs here. 1) lbound of plx ;  2) "where" condition